openapi-validator/src/main/java/org/ballerinalang/openapi/validator/TypeSymbolToJsonValidatorUtil.java

```diff
@@ -0,0 +1,438 @@
+            }
+            if (typeSymbol instanceof TypeReferenceTypeSymbol) {
+                typeSymbol = ((TypeReferenceTypeSymbol) typeSymbol).typeDescriptor();
+            }
+            isExitType = true;
+            List<ValidationError> validationError = validateRecordType((RecordTypeSymbol) typeSymbol, syntaxTree,
+                    semanticModel, properties, componentName, location);
+            validationErrorList.addAll(validationError);
+        } else if (typeSymbol instanceof StringTypeSymbol || typeSymbol instanceof IntTypeSymbol
+                || typeSymbol instanceof FloatTypeSymbol) {
+            if (!componentName.isBlank()) {
+                paramName = componentName;
+            } else {
+                paramName = "simpleParam";
+            }
+
+            if (typeSymbol.typeKind().getName().equals(convertOpenAPITypeToBallerina(schema.getType()))) {
+                // type mismatch
+                isExitType = true;
+            } else {
+                schemaType = schema.getType();
+                ballerinaType = typeSymbol.typeKind().getName();
+            }
+        } else if ((typeSymbol instanceof ArrayTypeSymbol) && (schema instanceof  ArraySchema)) {
+            if (!componentName.isBlank()) {
+                paramName = componentName;
+            } else {
+                paramName = "array";
+            }
+            if (((ArrayTypeSymbol) typeSymbol).memberTypeDescriptor().typeKind().getName().toLowerCase(Locale.ENGLISH)
+                    .equals(convertOpenAPITypeToBallerina(((ArraySchema) schema).getItems().getType()))) {
+                isExitType = true;
+            } else if ((((ArraySchema) schema).getItems() instanceof ObjectSchema) &&
+                    (((ArrayTypeSymbol) typeSymbol).memberTypeDescriptor() instanceof TypeReferenceTypeSymbol)) {
+                TypeSymbol recordType = null;
+                isExitType = true;
+                Optional<TypeSymbol> symbol = semanticModel
+                        .type(((ArrayTypeSymbol) typeSymbol).memberTypeDescriptor().location().lineRange());
+                if (symbol != null && symbol.isPresent()) {
+                    recordType = ((TypeDefinitionSymbol) symbol.get()).typeDescriptor();
+                }
+                List<ValidationError> recordValidationError = validate(((ArraySchema) schema).getItems(),
+                        recordType, syntaxTree, semanticModel, componentName, location);
+                validationErrorList.addAll(recordValidationError);
+
+            } else if ((((ArraySchema) schema).getItems() instanceof ArraySchema) &&
+                    (((ArrayTypeSymbol) typeSymbol).memberTypeDescriptor() instanceof ArrayTypeSymbol)) {
+                // handle nested array
+                isExitType = true;
+                ArrayTypeSymbol traverseNestedArray = (ArrayTypeSymbol) typeSymbol;
+                ArraySchema traversSchemaNestedArray = (ArraySchema) schema;
```

Comment:
Assignments are overriding inside if & these variables not used in the else part, So these assignments seem not needed.